src/addons/Portal/Portal.js

```diff
@@ -182,6 +200,15 @@ class Portal extends Component {
   // Component Event Handlers
   // ----------------------------------------
 
+  handlePortalClick = (e) => {
+    if (!this.props.closeOnCloseClick) return
+    if (!domUtils.closest(e.target, '[data-close]')) return
```

Comment:
I'm not sure of an alternative proposal yet, but I don't think we should be leaving the virtual DOM to accomplish this.  I see that this allows the user to define any element to be the "close" element and that this is important.

I'll give this some more thought and see what other proposals I can come up with.  I think the goal should be that we do not read from the real DOM though.